### PR TITLE
🐛 Source Hubspot: fix casting of float numbers

### LINKED ISSUE #10533: Update supported streams in Hubspot
### Tell us about the documentation you'd like us to add or update.
Ensure that [our documentation](https://docs.airbyte.com/integrations/sources/hubspot#supported-tables) accurately reflects the streams we have in [our connector](https://github.com/airbytehq/airbyte/tree/master/airbyte-integrations/connectors/source-hubspot/source_hubspot/schemas).

### PR DESCRIPTION
**Pull request by annalvova05**
## What
Close #10533 and #10463.

## How
Fix float numbers casting. Update documentation.

## Recommended reading order
1. `streams.py`
2. `hubspot.md`

## 🚨 User Impact 🚨
Are there any breaking changes? What is the end result perceived by the user? If yes, please merge this PR with the 🚨🚨 emoji so changelog authors can further highlight this if needed.

## Pre-merge Checklist
Expand the relevant checklist and delete the others.

<details><summary><strong>Updating a connector</strong></summary>

### Community member or Airbyter

- [ ] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret`
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)
- [ ] After the new connector version is published, connector version bumped in the seed directory as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
- [ ] Seed specs have been re-generated by building the platform and committing the changes to the seed spec files, as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)

</details>

### CODE REVIEW

File: airbyte-integrations/connectors/source-hubspot/source_hubspot/streams.py [hunk 1]
```diff
@@ -422,6 +422,7 @@ def _cast_value(cls, declared_field_types: List, field_name: str, field_value: A
         if target_type_name == "number":
             # do not cast numeric IDs into float, use integer instead
             target_type = int if field_name.endswith("_id") else target_type
```
Comment (bazarnov) [inline airbyte-integrations/connectors/source-hubspot/source_hubspot/streams.py:424]: Should it be `int` or `float`, based on [this](https://github.com/airbytehq/airbyte/issues/10463) issue?

**Review by bazarnov — COMMENTED** [review]
Could you please add some little `unit_test` to cover the [issue](https://github.com/airbytehq/airbyte/issues/10463) of the case?

File: airbyte-integrations/connectors/source-hubspot/source_hubspot/streams.py (hunk 1 above)
Comment (annalvova05) [reply airbyte-integrations/connectors/source-hubspot/source_hubspot/streams.py:424]: Should be float

**Review by sherifnada — APPROVED** [review]
@annalvova05 feel free to publish and merge once baz's comments are addressed!

**Review by bazarnov — APPROVED** [review]

**Pull request merged**